main.tf

```diff
@@ -1034,7 +1034,7 @@ resource "aws_route" "private_ipv6_egress" {
 
 locals {
   nat_gateway_count = var.single_nat_gateway ? 1 : var.one_nat_gateway_per_az ? length(var.azs) : local.max_subnet_length
-  nat_gateway_ips   = var.reuse_nat_ips ? var.external_nat_ip_ids : try(aws_eip.nat[*].id, [])
+  nat_gateway_ips   = var.reuse_nat_ips ? var.external_nat_ip_ids : aws_eip.nat[*].id
```

Comment:
why would this change make a difference?

Reply:
because the try statement can't be evaluated while planning, so the nat gateways are all replaced, despite the fact that they are really are not changed